operator/v1/types_console.go

```diff
@@ -99,8 +99,60 @@ type StatuspageProvider struct {
 	PageID string `json:"pageID"`
 }
 
+// CapabilityName defines name of UI capability in the console UI.
+type CapabilityName string
+
+const (
+	// lightspeedButton is the name for the Lightspeed button HTML element.
+	LightspeedButton CapabilityName = "LightspeedButton"
+)
+
+// CapabilityState defines the state of the capability in the console UI.
+type CapabilityState string
+
+const (
+	// "Enabled" means that the capability will be rendered in the console UI.
+	CapabilityEnabled CapabilityState = "Enabled"
+	// "Disabled" means that the capability will not be rendered in the console UI.
+	CapabilityDisabled CapabilityState = "Disabled"
+)
+
+// CapabilityVisibility defines the criteria to enable/disable a capability.
+// +union
+type CapabilityVisibility struct {
+	// state defines the if the capability is enabled or disabled in the console UI.
+	// Enabling the capability in the console UI is represented by the "Enable" value.
+	// Disabling the capability in the console UI is represented by the "Disabled" value.
+	// +unionDiscriminator
+	// +kubebuilder:validation:Enum:="Enabled";"Disabled"
+	// +kubebuilder:validation:Required
+	State CapabilityState `json:"state"`
+}
+
+// Capabilities contains set of UI capabilities and their state in the console UI.
+type Capability struct {
+	// name is the unique name of a capability.
+	// Available capabilities are LightspeedButton.
+	// +kubebuilder:validation:Enum:="LightspeedButton";
+	// +kubebuilder:validation:Required
+	Name CapabilityName `json:"name"`
+	// visibility defines the visibility state of capability.
+	// +kubebuilder:validation:Required
+	Visibility CapabilityVisibility `json:"visibility"`
+}
+
 // ConsoleCustomization defines a list of optional configuration for the console UI.
 type ConsoleCustomization struct {
+	// capabilities defines an array of capabilities that can be interacted with in the console UI.
+	// Each capability defines a visual state, by which it will get rendered.
+	// Available capabilities are LightspeedButton.
+	// Each of the available capabilities may appear only once in the list.
```

Comment:
What happens if you try and send a list containing two values with the same `name` value?

Reply:
this we are handling programatically in the console-operator.

Reply:
Yes, but I'm curious to see how that works in the API, it would be good to verify, if we remove the MaxItems, can you send the `name` twice and what will happen if you do

In theory the list type should handle this but I would like to verify it